src/main/scala/org/broadinstitute/dsde/workbench/sam/backgroundProcess/CryptominingSubscriber.scala

```diff
@@ -0,0 +1,47 @@
+package org.broadinstitute.dsde.workbench.sam
+package backgroundProcess
+
+import cats.effect.{ContextShift, IO}
+import fs2.Stream
+import io.chrisdavenport.log4cats.StructuredLogger
+import io.circe.Decoder
+import org.broadinstitute.dsde.workbench.google2.{Event, GoogleSubscriber}
+import org.broadinstitute.dsde.workbench.model.WorkbenchUserId
+import org.broadinstitute.dsde.workbench.sam.service.UserService
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+
+// Each instance of Sam will run a subscriber, since they all subscribe to the same topic via same subscription,
+// messages will be distributed among subscribers instead of distributed to all of them, which is what we want.
+class CryptominingSubscriber(subscriber: GoogleSubscriber[IO, CryptominingUserMessage],
```

Comment:
I don't think we should mention cryptomining. This is a general purpose mechanism for any automated external process to tell sam to disable a user. Cryptomining is just the first use of this.

Reply:
hmm..think my perspective is a bit different...this is a subscriber specifically for `terra-cryptomining` topic (which I was assuming we can potentially have other messages related to cryptomining activity other than disabling users), we can't really use this for another topic if there's another use case for disabling users.